internal/controller/etcdopstask/reconcile_task.go

```diff
@@ -0,0 +1,271 @@
+			return ctrlutils.ReconcileWithError(err)
+		}
+	} else {
+		logger.Info("Task state is not Pending, skipping transition", "currentState", task.Status.State)
+	}
+	return ctrlutils.ContinueReconcile()
+}
+
+// runTask executes the task handler's Run method.
+// It updates the task status based on the result of the run operation.
+// If the task is completed, it updates the LastOperation to completed or failed.
+// If the task is still in progress, it requeues the task for further processing.
+// If the task fails, it updates the task status to failed and sets the LastOperation to failed.
+// If the task succeeds, it updates the task status to succeeded and sets the LastOperation to completed.
+// If the task is already in a completed state, it skips the run operation.
+func (r *Reconciler) runTask(ctx context.Context, taskObjKey client.ObjectKey, taskHandler opsTask.Handler) ctrlutils.ReconcileStepResult {
+	_, err := r.getTask(ctx, taskObjKey)
+	if err != nil {
+		return ctrlutils.ReconcileWithError(err)
+	}
+	if err := r.recordLastOperation(ctx, taskObjKey, druidv1alpha1.OperationPhaseRunning, druidv1alpha1.OperationStateInProgress, ""); err != nil {
+		return ctrlutils.ReconcileWithError(err)
+	}
+	result := taskHandler.Run(ctx)
+	if result == nil {
+		err := druiderr.WrapError(
+			fmt.Errorf("run returned nil TaskResult; this is a bug in the TaskHandler implementation"),
+			ErrNilResult,
+			opsTask.RunOperation,
+			"run returned nil TaskResult; this is a bug in the TaskHandler implementation",
+		)
+		_ = r.recordLastError(ctx, taskObjKey, err)
+		return ctrlutils.ReconcileWithError(err)
+	}
+
+	if result.Completed {
+		if result.Error != nil {
+			// Task failed
+			err = r.recordLastError(ctx, taskObjKey, result.Error)
+			if err != nil {
+				return ctrlutils.ReconcileWithError(err)
+			}
+			if err := r.recordLastOperation(ctx, taskObjKey, druidv1alpha1.OperationPhaseRunning, druidv1alpha1.OperationStateFailed, result.Description); err != nil {
+				return ctrlutils.ReconcileWithError(err)
+			}
+			if err := r.recordTaskState(ctx, taskObjKey, druidv1alpha1.TaskStateFailed); err != nil {
+				return ctrlutils.ReconcileWithError(err)
+			}
+		} else {
+			// Task succeeded
+			if err := r.recordLastOperation(ctx, taskObjKey, druidv1alpha1.OperationPhaseRunning, druidv1alpha1.OperationStateCompleted, result.Description); err != nil {
+				return ctrlutils.ReconcileWithError(err)
+			}
+			if err := r.recordTaskState(ctx, taskObjKey, druidv1alpha1.TaskStateSucceeded); err != nil {
+				return ctrlutils.ReconcileWithError(err)
+			}
+		}
+
+		task, err := r.getTask(ctx, taskObjKey)
+		if err != nil {
+			return ctrlutils.ReconcileWithError(err)
+		}
+
+		if task.HasTTLExpired() {
+			return ctrlutils.ReconcileWithRequeue("Task completed, TTL expired, Requeueuing for cleanup")
+		}
+
+		return ctrlutils.ReconcileAfter(task.GetTimeToExpiry(), "Task completed, waiting for TTL to expire")
```

Comment:
I was wondering, why can't we just requeue with task.GetTimeToExpiry from the current task in memory. Do we need to fetch it again, in the same reconcile loop. Does it make so much of difference.